subprojects/core-api/src/main/java/org/gradle/api/Task.java

```diff
@@ -344,6 +344,24 @@ public String determineName(Task c) {
     @Incubating
     void onlyIf(String onlyIfReason, Spec<? super Task> onlyIfSpec);
 
+    /**
+     * <p>Execute the task only if the given provider returns true. The provider will be evaluated at task execution time, not
+     * during configuration. If the provider returns false, the task will be skipped.</p>
+     *
+     * <p>If given provider represents dependency to task like <code>TaskProvider.map { false }</code>, then such dependency
+     * will be added to this task accordingly.</p>
+     *
+     * <p>You may add multiple such predicates. The task is skipped if any of the predicates return false.</p>
+     *
+     * <p>Typical usage:<code>myTask.onlyIf(project.provider({ isProductionEnvironment() }))</code></p>
+     *
+     * @param onlyIfProvider provider to execute to determine if task should be run
+     *
+     * @since 8.2
+     */
+    @Incubating
+    void onlyIf(Provider<Boolean> onlyIfProvider);
```

Comment:
:question: Why not add the overload taking an `onlyIfReason` string?

Reply:
@eskatos Yeah, I'm going to add it also